desc/geometry/curve.py

```diff
@@ -740,10 +771,82 @@ def compute(
             transforms=transforms,
             data=data,
             override_grid=override_grid,
-            basis_in=self._basis,
+            basis_in=self.basis,
             **kwargs,
         )
 
+    @classmethod
+    def from_values(cls, coords, N=10, basis="xyz", name=""):
+        """Fit coordinates to FourierPlanarCurve representation.
+
+        Parameters
+        ----------
+        coords: ndarray, shape (num_coords,3)
+            Coordinates to fit a FourierPlanarCurve object with each column
+            corresponding to xyz or rpz depending on the basis argument.
+        N : int
+            Fourier resolution of the new r representation.
+        basis : {"rpz", "xyz"}
+            Basis for input coordinates. Defaults to "xyz".
+        name : str
+            Name for this curve.
+
+        Returns
+        -------
+        curve : FourierPlanarCurve
+            New representation of the curve parameterized by a Fourier series for r.
+
+        """
+        # convert to xyz basis
+        if basis == "rpz":
+            coords = rpz2xyz(coords)
+        coords = np.atleast_2d(coords)
+
+        # center
+        center = np.mean(coords, axis=0)
+        coords = coords - center  # shift to origin
+
+        # normal
+        idx = coords.shape[0] // 2
+        normal = np.cross(coords[0, :], coords[idx, :])  # 2 arbitrary points on curve
+        normal = normal / np.linalg.norm(normal)
+
+        # axis and angle of rotation
+        Z_axis = np.array([0, 0, 1])
+        axis = np.cross(Z_axis, normal)
+        axis = axis / np.linalg.norm(axis)
+        angle = np.arccos(np.dot(Z_axis, normal))
+        rotmat = rotation_matrix(axis, angle)
+        coords = coords @ rotmat  # rotate to X-Y plane
+        warnif(
+            np.max(np.abs(coords[:, 2])) > 1e-14,  # check that Z=0 for all points
+            UserWarning,
+            "Curve values are not planar!",
```

Comment:
Might want to give more info here, ie it still tries a least squares fit? or it takes the projection of the curve into a plane?

Reply:
It takes the projection into a plane. And now that I made your other requested change about using the average normal vector, that plane is the "best fit" in some sense.